Aprimoramento Dashboard Plotly Dash

Elias-Noda opened the pull request:
Esse PR coloca as modificações feita pelo auditor Renato em especial o aprimoramento do dashboard Plotly Dash.

Elias-Noda committed:
atualizando .gitignore

Elias-Noda committed:
Alterações do Renato

storopoli commented on analyse_data/regionalizar_distritos.py:
```diff
@@ -1,15 +1,21 @@
+from get_data import get_data, get_distritos, get_subprefeituras
 
 class RegionalizarDistritos:
 
```
Docstring?

h-pgy replied:
```suggestion
""" Teste docstring """
```

storopoli commented on analyse_data/regionalizar_distritos.py:
```diff
@@ -1,15 +1,21 @@
 class RegionalizarDistritos:
 
-
     def agrupar_distritos_media(self, microdados_ideb):
```
Docstring?

storopoli commented on analyse_data/regionalizar_distritos.py:
```diff
@@ -1,15 +1,21 @@
 
         return grouped
 
+    def agrupar_subprefeituras_media(self, microdados_ideb):
```
Docstring?

h-pgy replied:
```suggestion
    def agrupar_subprefeituras_media(self, microdados_ideb):
    '''Teste de docstring. '''
```

storopoli commented on analyse_data/regionalizar_distritos.py:
```diff
@@ -1,15 +1,21 @@
+
+        return grouped
+
     def filtrar_tipo_anos(self, grouped, tipo_anos):
```
Docstring?

storopoli commented on analyse_data/regionalizar_distritos.py:
```diff
@@ -23,6 +29,18 @@ def padronizar_codigo_distrito_ideb(self, df_anos):
+
+        return df_anos
+
+    def padronizar_codigo_subprefeitura_ideb(self, df_anos):
```
Docstring?

storopoli commented on analyse_data/regionalizar_distritos.py:
```diff
@@ -23,6 +29,18 @@ def padronizar_codigo_distrito_ideb(self, df_anos):
+
         return df_anos
 
     def merge_shapefile(self, df_anos, distritos):
```
Docstring?

storopoli commented on analyse_data/regionalizar_distritos.py:
```diff
@@ -34,9 +52,18 @@ def merge_shapefile(self, df_anos, distritos):
-    
 
-    def __call__(self, tipo_anos, microdados_ideb=None, distritos=None):
+    def merge_shapefile_sub(self, df_anos, subprefeituras):
```
Docstring?

storopoli commented on analyse_data/regionalizar_distritos.py:
```diff
@@ -34,9 +52,18 @@ def merge_shapefile(self, df_anos, distritos):
+        return merged
+
+
+    def __call__(self, tipo_anos, microdados_ideb=None, distritos=None, subprefeituras=None):
```
Docstring?

storopoli commented on app.py:
```diff
@@ -1,93 +1,1123 @@
 
-regionalizar = RegionalizarDistritos()
 
 def gerar_geodf(anos_ideb):
```
Docstring?

storopoli commented on app.py:
```diff
@@ -1,93 +1,1123 @@
-    ))
-    fig.update_geos(fitbounds="locations", visible=False)
-    fig.update_layout(margin={"r":0,"t":20,"l":0,"b":0})
+def gerar_texto():
```
Docstring?

storopoli commented on app.py:
```diff
@@ -1,93 +1,1123 @@
+
+    return totais
+
+def gerar_mapa(tipografico, anos_ideb, tipodados):
```
Docstring?

storopoli commented on app.py:
```diff
@@ -1,93 +1,1123 @@
+
+
+
+def displayMapa(indicadores, dados, anos):
```
Docstring?

storopoli commented on app.py:
```diff
@@ -1,93 +1,1123 @@
+
+
+
+def toggle_modal(n1, n2, is_open):
```
Docstring?

storopoli commented on get_data/__init__.py:
```diff
@@ -1,4 +1,4 @@
+from .cached_data import merged_data, distritos, subprefeituras
 from .merge_cadastro_ideb import JoinData
 
 def get_data():
```
Docstring?

storopoli commented on get_data/__init__.py:
```diff
@@ -11,4 +11,10 @@ def get_distritos():
 
+    return geodf
+
+def get_subprefeituras():
```
Docstring?

storopoli commented on get_data/cached_data.py:
```diff
@@ -95,5 +95,18 @@ def distritos():
 
     return geodf
 
+def subprefeituras():
```
Docstring?

storopoli commented on get_data/distritos_shp.py:
```diff
@@ -4,6 +4,9 @@
+
+
+
 class DownloadShapeDists:
```
Docstring?

storopoli commented on get_data/obterDistritos.py:
```diff
@@ -0,0 +1,52 @@
+from json import dumps
+
+
+def distritos(path_salvo, dfIdebIniciais, dfIdebFinais):
```
Docstring?

storopoli commented on get_data/obterIdeb.py:
```diff
@@ -0,0 +1,16 @@
+import pandas as pd
+
+
+def dadosIdeb(path_salvo):
```
Docstring?

storopoli commented on get_data/obterSubprefeituras.py:
```diff
@@ -0,0 +1,46 @@
+import geopandas as gpd
+
+
+def subprefeituras(path_salvo, dfIdebIniciais, dfIdebFinais):
```
Docstring?

storopoli commented on mapa.py:
```diff
@@ -0,0 +1,103 @@
+from analyse_data.regionalizar_distritos import RegionalizarDistritos
+from get_data import get_data, get_distritos, get_subprefeituras, obterIdeb, obterDistritos, obterSubprefeituras
+
+dfDadosIdeb = obterIdeb.dadosIdeb('data/cadastro_ideb_merged.csv')
```
Não dá para tudo isso aqui solto virar uma classe/função?

storopoli commented on mapa.py:
```diff
@@ -0,0 +1,103 @@
+@app.callback(
+    Output("choropleth", "figure"),
+    [Input("candidate", "value")])
+def display_choropleth(candidate):
```
Docstring?

storopoli commented on mapa2.py:
```diff
@@ -0,0 +1,56 @@
+     ],
+    Input('dropdown1','value')
+)
+def update_map(county_select):
```
Docstring?

storopoli commented on mapa2.py:
```diff
@@ -0,0 +1,56 @@
+import dash_html_components as html
+from dash.dependencies import Input, Output
+
+with urlopen('https://raw.githubusercontent.com/plotly/datasets/master/geojson-counties-fips.json') as response:
```
Não dá para tudo isso aqui solto virar uma classe/função?

storopoli commented on app.py:
```diff
@@ -1,93 +1,1123 @@
+from get_data import get_data, get_distritos, get_subprefeituras, obterIdeb, obterDistritos, obterSubprefeituras
+import dash_table
+
+dfDadosIdeb = obterIdeb.dadosIdeb('data/cadastro_ideb_merged.csv')
```
Não dá para tudo isso aqui solto virar uma classe/função?

storopoli reviewed (changes requested):
Precisa documentar tudo isso e tem umas coisas soltas no começo do `app.py` e `mapa.py` que dá para colocar como função/classe/

h-pgy reviewed (changes requested):
testes

Elias-Noda committed:
Adicionando dropdowns dinâmicos e modo escuro

Está versão ainda não está completa e possui alguns bugs

Elias-Noda committed:
Atualização menor

Melhorando adicionando algumas cores novas ao tema escuro para melhor vizualização

Removendo dropdowns dinâmicos para concertar alguns bugs

Elias-Noda committed:
Atualizações estéticas

Melhoramento estético do modo escuro com a adição de um external stylesheet diferente

Adição de algumas linhas de código e comentários para auxiliar na rápida alternação entre modo claro e escuro

Elias-Noda committed:
Gráficos de Linha - Universalização

Adicionado gráficos de linhas para o indicador de Universalização

Elias-Noda committed:
Atualização menor

Elias-Noda committed:
Atualização modo-escuro

Concertando bugs no modo escuro dos dropdowns

IMPORTANTE: O mapa em universalização é apenas um teste

Elias-Noda committed:
Concerto de bugs

Adicionando espaçamento no começo e fim da página para melhor vizualização estética

Colocando gráfico de barras em ordem alfabética e concertando um bug de nem todos os nomes estarem sendo mostrados

Elias-Noda committed:
Novas alterações do Renato

Adição de gráficos e algumas outras modificações

Elias-Noda committed:
Correção de bugs

Arrumando o problema com os dataframes que geravam mensagens de erro ao rodar o programa

Melhorando a visibilidade do código, em especial concertando todos os weak warnings possíveis sem desestruturar o fluxo do programa

Concertando os medium warnings possíveis sem causar instabilidade no programa

Elias-Noda committed:
Atualizando app.py

Adicionando docstrings ao app.py

Com todas as atualizações recentes, muitas partes do programa se tornaram obsoletas, por isso primeiro removerei essas partes redundantes do código antes de aplicar docstrings aos outros .py, para evitar informações desconexas nas mesmas.

Elias-Noda committed:
Merge pull request #1 from ObservatorioPoliticasTCM/version_test

Merge de version_test

Elias-Noda committed:
Docstrings e função lambda

Eliminando algumas partes obsoletas do código e terminando de aplicar as docstrings

Aplicando a função lambda nas colunas de gastos_2019 e universalização_2019 e universalizacao_2020 do cadastro_merged_ideb.csv

Elias-Noda committed:
Merge pull request #2 from ObservatorioPoliticasTCM/version_test

Docstrings e função lambda
